signature/src/image/mod.rs

```diff
@@ -0,0 +1,111 @@
+// Copyright (c) 2021 Alibaba Cloud
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+use oci_distribution::Reference;
+
+pub mod digest;
+
+use digest::Digest;
+
+#[derive(EnumString, Display, Debug, PartialEq)]
+pub enum TransportName {
+    #[strum(to_string = "docker")]
+    Docker,
+    #[strum(to_string = "dir")]
+    Dir
+}
+
+// Image contains information about the image which may be used in signature verification.
+pub struct Image {
+    pub reference: Reference,
+    // digest format: "digest-algo:digest-value"
+    pub manifest_digest: Digest,
+
+    // Try signature extensions in registry before using the sigstore.
+    // (https://github.com/containers/image/issues/384)
+    //
+    // FIXME: now don't support, always set to false.
+    // 
+    // issue: https://github.com/confidential-containers/image-rs/issues/12
+    pub registry_xrss_api_support: bool,
+}
+
+impl Image {
+    pub fn default_with_reference(image_ref: Reference) -> Self {
+        Image {
+            reference: image_ref,
+            manifest_digest: Digest::default(),
+            registry_xrss_api_support: false,
+        }
+    }
+
+    pub fn transport_name(&self) -> String {
+        // FIXME: Now only support "docker" transport (and it is hardcoded).
+        // TODO: support "dir" transport.
+        // 
+        // issue: https://github.com/confidential-containers/image-rs/issues/11
+        TransportName::Docker.to_string()
+    }
+
+    pub fn set_manifest_digest(&mut self, digest: &str) -> Result<()> {
+        self.manifest_digest = Digest::try_from(digest)?;
+    }
+}
+
+// Get repository full name:
+// `registry-name/repository-name`
+pub fn get_image_repository_full_name(image_ref: &Reference) -> String {
+    if image_ref.registry().is_empty() {
+        image_ref.repository().to_string()
+    } else {
+        format!("{}/{}", image_ref.registry(), image_ref.repository())
+    }
+}
+
+// Returns a list of other policy configuration namespaces to search.
```

Comment:
Yes, because `docker.io/library/busybox:latest` or` docker.io/library` or `docker.io` may be constrained in the policy file, and for the image `docker.io/library/busybox:latest`,no matter which of the three cases, it needs to meet the policy configuration in it.

Maybe you can refer to this: https://github.com/containers/image/blob/main/docs/containers-policy.json.5.md#docker